### PR TITLE
Added test to show how to use IOProblem

### PR DESCRIPTION
**Pull request by whiskeysierra**
See #36

### CODE REVIEW

File: jackson-datatype-problem/src/test/java/org/zalando/problem/IOProblem.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Optional;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
```
Comment (jhorstmann) [inline jackson-datatype-problem/src/test/java/org/zalando/problem/IOProblem.java]: So this is the trick to make deserialization work correctly?

File: jackson-datatype-problem/src/test/java/org/zalando/problem/IOProblem.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        return detail;
+    }
+
+    @Override
+    public Optional<URI> getInstance() {
+        return instance;
+    }
+
+    @Override
+    public ThrowableProblem getCause() {
```
Comment (jhorstmann) [inline jackson-datatype-problem/src/test/java/org/zalando/problem/IOProblem.java]: Could `getCause` be a default method in `Exceptional`? Having to override it just to return null is not really intuitive.

File: jackson-datatype-problem/src/test/java/org/zalando/problem/IOProblem.java (hunk 1 above)
Comment (jhorstmann) [inline jackson-datatype-problem/src/test/java/org/zalando/problem/IOProblem.java]: > So this is the trick to make deserialization work correctly?

Ok, it is, I saw your comment on issue #36.

File: jackson-datatype-problem/src/test/java/org/zalando/problem/IOProblem.java (hunk 2 above)
Comment (whiskeysierra) [inline jackson-datatype-problem/src/test/java/org/zalando/problem/IOProblem.java]: > Could getCause be a default method in Exceptional? Having to override it just to return null is not really intuitive.

We could, but it won't fix the issue. Default methods will *step down* if there is a method from a class with the same signature and `Throwable.getClause` matches that. We could drop the explicit return type (from ThrowableProblem to Throwable) though.

**Pull request merged**